libs/csm/rest/csm_rest_query_deployment.py

```diff
@@ -0,0 +1,183 @@
+        """
+        Get node topology
+        :param header: header for api authentication
+        :return: response
+        """
+        self.log.info("Get node topology request....")
+        query_param = cluster_id + '/' + 'node'
+        if node_id is not None:
+            query_param = query_param + '/' + node_id
+        response = self.get_cluster_topology(auth_header, query_param)
+        return response
+
+    def verify_nodes(self, expected_response=HTTPStatus.OK):
+        """
+        Verify number of nodes and node names
+        """
+        resp = self.get_storage_topology()
+        solution_yaml = self.get_solution_yaml()
+        result = resp.status_code == expected_response
+        if result:
+            get_response = resp.json()
+            self.log.info("Verify node names")
+            self.log.info("Verify number of nodes in resp and solution yaml")
+            nodes = len(get_response["topology"]["cluster"]["nodes"])
+            num_nodes = len(solution_yaml["solution"]["storage_set"]["nodes"])
+            if num_nodes != nodes:
+                self.log.error("Actual and expected response for number of nodes didnt match")
+                result = False
+        else:
+            self.log.error("Status code check failed.")
+        return result
+
+    def verify_storage_set(self, storage_set_id: str = None, expected_response=HTTPStatus.OK):
+        """
+        Verify storage set details
+        """
+        resp = self.get_storage_topology(storage_set_id)
+        solution_yaml = self.get_solution_yaml()
+        result = resp.status_code == expected_response
+        if result:
+            get_response = resp.json()
+            self.log.info("Verifying sns and dix values in resp and solution yaml")
+            storage_set_dix = get_response[
+                "topology"]["cluster"]["storage_set"]["durability"]["dix"]
+            resp_dix = resp_dix.join([str(storage_set_dix["data"]), str(storage_set_dix["parity"]),
+                                      str(storage_set_dix["spare"])])
+            input_dix = solution_yaml["solution"]["storage_set"]["durability"]["dix"]
+            if input_dix != resp_dix:
+                self.log.error("Actual and expected response for dix didn't match")
+                result = False
+            storage_set_sns = get_response[
+                "topology"]["cluster"]["storage_set"]["durability"]["sns"]
+            resp_sns = resp_sns.join([str(storage_set_sns["data"]), str(storage_set_sns["parity"]),
+                                      str(storage_set_sns["spare"])])
+            input_sns = solution_yaml["solution"]["storage_set"]["durability"]["sns"]
+            if input_sns != resp_sns:
+                self.log.error("Actual and expected response for sns didnt match")
+                result = False
+        else:
+            self.log.error("Status code check failed.")
+            result = False
+        return resp, result
```

Comment:
Can u prepare a error message and send it to test for assertion

return resp, result, err_msg

Reply:
done